mitocw/edx-platform
# Enroll CCX coaches in the courses to which they're added

Hi EdX,

First PR for me -- I'm a new member of the MIT ODL engineering team, so I think I'm covered by our institutional agreement. This one is a small change that addresses a rough edge of CCX. 

**Motivation:** MIT ODL issue https://github.com/mitocw/edx-platform/issues/37. A user who is registered but not enrolled in the course, when added as a CCX coach, is not enrolled, and therefore can't see the courseware which s/he is the coach of. In other words, users must be both enrolled manually and added as CCX coaches manually. This PR makes that one step.

**Changes**: Only LMS is affected by this change, and apart from the behavior described below, under "How to test", the only user-visible effect is that the CCX coach gets an email notifying her/him that they've been enrolled in the course. For an instructor adding a coach, there is now one fewer step. This PR makes no change to the behavior of LMS in the case that the user being added is not registered at all.

**How to test**:
[First, enable CCX](http://edx-installing-configuring-and-running.readthedocs.org/en/latest/configuration/enable_ccx.html).

*Before checking out this PR (ie, while running edx-platform master):*

1. Visit the instructor dashboard, (eg: ```/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-membership``)
1. From the "Course Team Management" block, select "CCX Coaches": 
![image](https://cloud.githubusercontent.com/assets/882377/10201819/e720cea0-677b-11e5-82f6-d58c2a53949a.png)
1. Enter the username or email of a user who is *not* currently enrolled in the course:
![image](https://cloud.githubusercontent.com/assets/882377/10199750/5d640ea8-6770-11e5-890b-931c9c7d12b9.png)
1. Note that if you then attempt to unenroll that same user, you get a message, indicating that the user is not enrolled in the course:
![image](https://cloud.githubusercontent.com/assets/882377/10201828/f1cf4250-677b-11e5-8a0d-9cde1ac267ab.png)

*After checking out this PR:*

1. Repeat steps 1 and 2 above
1. Remove that user as a CCX coach.
1. Add that user as a CCX coach.
1. Unenroll that user from the course, and you will see that the user's been unenrolled successfully, indicating that the user was enrolled in the process of being added as a CCX coach:
![image](https://cloud.githubusercontent.com/assets/882377/10201835/f86c562a-677b-11e5-9ddd-ca4286814307.png)

Fixes https://github.com/mitocw/edx-platform/issues/37
